### ISSUE TITLE
World browser pagination not apparent enough

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#1117]
Various users have provided feedback on the world browser's pagination. Often, users have not realized there were multiple pages in their search query, e.g. users not immediately recognizing there was a second page of the MMC category.

There are a few potential tweaks/approaches that could be made in the future to improve visibility of pagination, and the searching experience:

- The page counter could return the total number of results for the search query, e.g. "1-50 (of 81)" for the MMC category
- The visibility of the next/previous page could be increased either via some form of coloration/additional contrast, or the use of a localizable/translatable string, e.g. "Next Page/Previous Page" instead of symbols
- The search results could potentially support 'infinite scrolling' with the pagination serving as a quick shortcut to navigate the results

I believe a part of the underlying issue/confusion is that more than one page's worth of results are displayed by default on the world browser, e.g. the user has to scroll to get to the bottom of the list, as such, when they reach the bottom, and no additional results are shown, they might assume at that point they've reached the end of the list without realizing there is another page.

As such, another potential solution could be:
- Restricting the results to display only what can fit on a single non-scrollable page, and use pagination exclusively

**Comment by Frooxius** [comment]
I could see about increasing the contrast of visibility of the next/previous page button, so it's easier to tell when it's on and off. One thing that could be added is "..." at the end if there are more pages. E.g. "1-50..."

- Doing "(of 81)" is not technically possible, because the total number of entries isn't actually known. They're continuously queried only when you actually switch to another page.
- I don't think "Next Page/Previous Page" would help. I don't think the issue stems from people not understanding what the buttons are and this would just take significantly more space.
- Infinite scrolling could be a potential addition, but that's a pretty major one and would take a while.
- I don't think restricting results to non-scrollable page is a good solution either, that would make it harder to interact with and also harder to technically enforce - number of displayed entries will depend on the current size of the facet.

The best thing I feel would be adding the "..." at the end of numbers and perhaps changing the symbols of the buttons a bit. E.g. if  page is available display the arrow, otherwise display some other symbol to indicate it's the end, maybe an empty arrow. Adding some color coding could help too (for color-blind the change of symbol should help).

**Comment by shiftyscales** [comment]
Yeah- any bit could help of the solutions you proposed, @Frooxius. Just wanted to clarify that this isn't a pressing/urgent issue- it is something that can be taken care of at a later time.

It is also something that could potentially by handled by @Coffee-Maker via additional polish of the visual design.

**Comment by H3BO3** [comment]
I suggested putting previous page/next page buttons at the bottom of the scrollable area, so they only come into view when the bottom of the page is reached. The current page switcher is fine design-wise, it's just that visibility is limited due to its size and location. A bit of redundancy with prev/next buttons that only become visible at the end of the page would solve the problem of page awareness/visibility without making things busier overall.
